### PR TITLE
 Older version of AzCopy appends version check error message to piped download

### PR DESCRIPTION
**Pull request by dphulkar-msft**
Bug: https://msazure.visualstudio.com/One/_workitems/edit/28689197
Tested:
SS1

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -267,6 +273,8 @@ func init() {
 	_ = rootCmd.PersistentFlags().MarkHidden("await-continue")
 	_ = rootCmd.PersistentFlags().MarkHidden("await-open")
 	_ = rootCmd.PersistentFlags().MarkHidden("debug-skip-files")
+
+	rootCmd.PersistentFlags().StringVar(&azcopyFromToFlag, "from-to", "", "Used to check if from-to is BlobPipe type then avoid version check as it appends version update info log which causes failure in download.")
```
Comment (vibhansa-msft) [inline cmd/root.go:277]: Will this create a new flag as the from-to will be there already defined in some other file.

Comment (dphulkar-msft) [reply cmd/root.go]: updated. using GetString() to retrieve the flag value

**Review by adreed-msft — APPROVED** [review]

File: e2etest/zt_newe2e_s2s_test.go [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
+	"strconv"
+
 	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/blob"
 	"github.com/Azure/azure-storage-azcopy/v10/common"
-	"strconv"
 )
 
 func init() {
-	suiteManager.RegisterSuite(&S2STestSuite{})
+	//suiteManager.RegisterSuite(&S2STestSuite{})
```
Comment (vibhansa-msft) [inline e2etest/zt_newe2e_s2s_test.go:11]: Why are these commented in all the files

File: e2etest/zt_newe2e_blob_test.go [hunk 3]
```diff
@@ -12,136 +10,200 @@ func init() {
...
-			},
-		})
-
-	ValidateResource[ObjectResourceManager](svm, dstObj, ResourceDefinitionObject{
-		Body: body,
-	}, true)
-}
-
-func (s *BlobTestSuite) Scenario_DownloadBlob(svm *ScenarioVariationManager) {
+// func (s *BlobTestSuite) Scenario_UploadBlockBlobs(svm *ScenarioVariationManager) {
```
Comment (vibhansa-msft) [inline e2etest/zt_newe2e_blob_test.go:13]: Why are all these test cases commented here?

File: e2etest/zt_newe2e_s2s_test.go (hunk 2 above)
Comment (dphulkar-msft) [reply e2etest/zt_newe2e_s2s_test.go:11]: commented as of now. To validate the new test case added.
Will remove these changes and create a new ticket for adding a TC for this scenario.

File: e2etest/zt_newe2e_blob_test.go (hunk 3 above)
Comment (dphulkar-msft) [reply e2etest/zt_newe2e_blob_test.go:13]: commented as of now. To validate the new test case added.
Will remove these changes and create a new ticket for adding a TC for this scenario.

File: e2etest/zt_newe2e_blob_test.go [hunk 4]
```diff
@@ -204,3 +205,31 @@ func (s *BlobTestSuite) Scenario_DownloadBlobRecursive(svm *ScenarioVariationMan
...
+		Body:       body,
+	})
+
+	stdout, _ := RunAzCopy(
+		svm,
+		AzCopyCommand{
+			Verb: AzCopyVerbCopy,
+			Targets: []ResourceManager{
+				TryApplySpecificAuthType(srcObj, ResolveVariation(svm, []ExplicitCredentialTypes{EExplicitCredentialType.SASToken()}), svm, CreateAzCopyTargetOptions{}),
+				dstContainer,
```
[the remark this answers is not in the document]
Comment (dphulkar-msft) [reply e2etest/zt_newe2e_blob_test.go]: test case addition will be a part of different PR.
Resolving the conversation

File: cmd/root.go [hunk 5]
```diff
@@ -136,6 +137,18 @@ var rootCmd = &cobra.Command{
...
 
+		var fromToFlagValue string
+		if cmd.Flags().Changed("from-to") {
+			// Access the value of the "from-to" flag
+			fromToFlagValue, err = cmd.Flags().GetString("from-to")
+			if err != nil {
+				return fmt.Errorf("error accessing 'from-to' flag: %v", err)
+			}
+		}
+		if fromToFlagValue == "BlobPipe" {
```
[the remark this answers is not in the document]
Comment (dphulkar-msft) [reply cmd/root.go]: makes sense. added inside if block

**Review by vibhansa-msft — APPROVED** [review]

**Comment by seanmcc-msft** [comment]
@dphulkar-msft, please add a test for this case.

**Pull request merged**